jax/_src/scipy/sparse/linalg.py

```diff
@@ -73,6 +73,15 @@ def _identity(x):
   return x
 
 
+def _normalize_matvec(f):
+  """Normalize an argument for computing matrix-vector products."""
+  if callable(f):
+    return f
+  else:
```

Comment:
WDYT about something like

```python
if callable(f):
  return f
elif isinstance(f, jax.numpy.ndarray):
  return partial(_dot, f)
else:
  raise TypeError(...)
```

Basically, do a bit more input validation here, if it's easy. Could also check that it's 2D and square.

Reply:
Good idea, done!